### ISSUE TITLE
About new gazebo_map code

### ISSUE THREAD

**Issue by KkCabin** [engcang/gazebo_maps#2]
Hello, I looked at your latest open source work [FAST_LIO_MULTI](https://github.com/engcang/FAST_LIO_MULTI) and thank you for your excellent work. I watched your Youtube video and it is so cool, I have two questions for you. 
The first question is that I observed some slight discrepancies in both the two different ways of using multiple lasers and groundtruth, is this a problem in the single laser case?
The second question is, I wonder if you are comfortable open sourcing Gazebo simulations? I'm really looking forward to trying it out myself.

**Comment by engcang** [comment]
Hi @KkCabin ! Thank you for your interests!
For the second question, yes I will open the simulation few months later due to NDA issue of the project! so no for right now.
For the first question, I am sorry but I didn't get what you are asking exactly :) can you repeat the question please?

**Comment by KkCabin** [comment]
> Hi @KkCabin ! Thank you for your interests! For the second question, yes I will open the simulation few months later due to NDA issue of the project! so no for right now. For the first question, I am sorry but I didn't get what you are asking exactly :) can you repeat the question please?

Hello, thanks for your reply. Sorry that I didn't express the problem clearly. Let me rephrase the problem.
I watched the video you posted on youtube and your work is great. But I noticed that whether it is Bundle or Async method, they do not completely coincide with groundtruth. I would like to ask if this small drift is caused by adding multiple lasers? Or is it a problem with FAST-LIO itself? Not sure if I made it clear.

**Comment by KkCabin** [comment]
I don't know if you have tried to compare the effects of single laser and multiple lasers.
I would like to ask you this question because I also tried to modify the FAST-LIO algorithm so that it can use multiple lasers. The method I use is similar to your sync method. I applied the modified algorithm to my platform, but there was no data to use as ground truth. I am currently using a gnss trajectory as ground truth. The current effect is that there is also a small drift between the trajectory output by FAST-LIO and the trajectory of gnss. I can't figure out what is causing this drift. I guess there may be four reasons:
1. There is a problem with my algorithm modification.
2. The calibration between lasers is not completely accurate.
3. FAST-LIO itself is not completely accurate.
4. gnss is not suitable as a true value.
But I don't know how to determine where the problem is, and I would like to ask you for advice.

**Comment by KkCabin** [comment]
Furthermore, I would like to ask if you have tried to compare the effects of single laser and multiple lasers.
I would like to ask you this question because I also tried to modify the FAST-LIO algorithm so that it can use multiple lasers. The method I use is similar to your Bundle method. I applied the modified algorithm to my platform, but there was no data to use as ground truth. I am currently using a gnss trajectory as ground truth. The current effect is that there is also a small drift between the trajectory output by FAST-LIO and the trajectory of gnss. I can't figure out what is causing this drift. I guess there may be four reasons:
1. There is a problem with my algorithm modification.
2. The calibration between lasers is not completely accurate.
3. FAST-LIO itself is not completely accurate.
4. gnss is not suitable as a true value.
But I don't know how to determine where the problem is, and I would like to ask you for advice.

**Comment by engcang** [comment]
Hi @KkCabin . Thank you for your detailed explanation!
I think this is the problem of FAST-LIO itself. It matches current LiDAR scans to saved ikd-tree map to update the filter with point-to-plane measurement matrix. If point-to-plane matching cannot be done properly, the pose estimation gets worse. 
For my simulations, I deliberately flew the drone around the textureless (or free of objects) environments to make point-to-plane matching failed and test the robustness of the update methods. That is why the results do not completely coincide with GT.

The same goes for your case, I guess. But that depends on your sensor configuration and the environments where you are scanning. 
I also tried single LiDAR for the same simulation bag file, and the result for the first sequence got worse and for the second sequence, FAST-LIO diverges because there is not enough scanned data to perform point-to-plane matching.

**Comment by KkCabin** [comment]
Hi @engcang.
Thank you for your detailed sharing :) , all my questions have been resolved, thank you! I think it's time to further modify FAST-LIO to suit my environment.
